CONSOLE-3824: Remove kubevirt-plugin code

pcbailey opened the pull request:
This PR does the following:

- Removes the kubevirt-plugin package and references in configuration files
- Copies code upon which the topology plugin is dependent from the kubevirt-plugin package to the topology package
- Changed the i18n prefix in the network-attachment-definition-plugin package from `kubevirt-plugin` to `network-attachment-definition-plugin`

Jira: https://issues.redhat.com/browse/CONSOLE-3824

avivtur commented:
This PR will also solve: https://issues.redhat.com//browse/CNV-37607

pcbailey commented:
/retest

pcbailey commented:
/retest

avivtur commented:
Excellent work @pcbailey :1st_place_medal: 
/lgtm

jhadvig commented:
QE Approver:
/assign @yapei 
Docs Approver:
/assign @opayne1 
PX Approver:
/assign @RickJWagner

RickJWagner commented:
/label px-approved

yapei commented on frontend/webpack.config.ts:
```diff
@@ -326,12 +326,14 @@ const config: Configuration = {
     new CopyWebpackPlugin([
       { from: './packages/network-attachment-definition-plugin/locales', to: 'locales' },
     ]),
+    new CopyWebpackPlugin([
+      { from: './packages/network-attachment-definition-plugin/locales', to: 'locales' },
+    ]),
```
are these lines duplicate?

pcbailey replied:
Yes, they were. I removed the duplicate. Thanks for catching that!

yapei commented:
1. launched a cluster, subscribed `OpenShift Virtualization` operator successfully
2. Verify `Virtualization` menu exists
3. Verify if secondary menu `Network Attachment Definitions` exist in `Networking` menu
-> No `Network Attachment Definitions` menu, see screen recording, not sure if this is in the scope of this PR

https://github.com/openshift/console/assets/12692381/0c1af56c-3e3b-4af8-808a-d5ed7f6618e9

yapei commented:
also due to https://issues.redhat.com/browse/OCPBUGS-22853 I can't verify if VM topology codes are working as expected or not(moving away `frontend/packages/kubevirt-plugin/src/topology` to `frontend/packages/topology` )

opayne1 commented on frontend/packages/network-attachment-definition-plugin/locales/en/network-attachment-definition-plugin.json:
```diff
@@ -0,0 +1,32 @@
+  "Bridge name": "Bridge name",
+  "MAC spoof check": "MAC spoof check",
+  "Bridge mapping": "Bridge mapping",
+  "Physical network name. A bridge mapping must be configured on cluster nodes to map between physical network names and Open vSwitch bridges.": "Physical network name. A bridge mapping must be configured on cluster nodes to map between physical network names and Open vSwitch bridges.",
```
```suggestion
  "Physical network name. Bridge mapping must be configured on cluster nodes to map between physical network names and Open vSwitch bridges.": "Physical network name. Bridge mapping must be configured on cluster nodes to map between physical network names and Open vSwitch bridges.",
```

pcbailey replied:
Updated! Thank you!

opayne1 reviewed (commented):
One small comment but otherwise looks good

/label docs-approved

pcbailey commented:
> 1. launched a cluster, subscribed `OpenShift Virtualization` operator successfully
> 2. Verify `Virtualization` menu exists
> 3. Verify if secondary menu `Network Attachment Definitions` exist in `Networking` menu
>    -> No `Network Attachment Definitions` menu, see screen recording

I fixed this issue. It was looking for the `KUBEVIRT` flag, so I updated it to look for the `KUBEVIRT_DYNAMIC` flag.

pcbailey commented:
> also due to https://issues.redhat.com/browse/OCPBUGS-22853 I can't verify if VM topology codes are working as expected or not(these codes moved away from `frontend/packages/kubevirt-plugin/src/topology` to `frontend/packages/topology` ) since `VirtualMachines` page is not accessible

I'm not sure I understand what you mean. That bug pertains to OCP 4.14/CNV 4.13. I also confirmed that the root cause of that bug is no longer present in our code from 4.14 onward. I was able to test the Topology page and Virtual Machines pages. Could you please clarify the issue you mention here? Thanks!

pcbailey commented:
/retest

pcbailey commented:
/retest

pcbailey commented:
/retest

avivtur commented:
/lgtm

pcbailey commented:
/retest

pcbailey commented:
/retest

pcbailey commented:
/retest

pcbailey commented:
/retest

yapei commented:
I still can't visit VirtualMachines page and other pages(OpenShift Virtualization 4.14.0 installed), here is the recording

https://github.com/openshift/console/assets/12692381/8088a280-81bf-46fd-8be6-22e935d30231

pcbailey commented:
> I still can't visit VirtualMachines page and other pages(OpenShift Virtualization 4.14.0 installed), here is the recording

Ah, I understand now. This PR should be tested on the latest versions of OCP and OpenShift Virtualization.

yapei commented:
generally the changes look good from QE's point of view
/label qe-approved

pcbailey commented:
/retest

jhadvig reviewed (approved):
/approve
